functions/api/branch.js

```diff
@@ -173,7 +173,7 @@ router.put('/', isAdmin, async (request, response, next) => {
  * @description Route to delete a branch
  * @throws 400 if branch does not exist
  */
-router.delete('/:id', isAdmin, async (request, response, next) => {
+router.delete('/:id', isAdminOrSuperAdmin, async (request, response, next) => {
```

Comment:
For delete we are going to change the functionality, right? For this delete , I think it should check for isSuperAdmin.
I think we should have separate route for updating isActive flag since now it is now a very important update. We may need to update triggers. Let me know what you think.

Reply:
umm, yes. I havent taken care of other routes like branch,category... delete operation. Yes, it should be "isSuperAdmin".

Reply:
Isnt triggers based on db operations. Like .document(`/${constants.BRANCHES}/{branch}/${constants.TRANSACTIONS}/{transactionId}`)
So how does it matter if if we make inactive separate route? For inactivation, if we need to update triggers, we should check for oldData.isActive !== newData.isActive and do necessary action. Let me know if I've missed something. Also, I think UI agreed to use same PUT flow to update to "inactive" as well.

Reply:
updated **isSuperAdmin** for other api's as well.